From ceb85e7fb83fdab0fd4ddba439912e53278922ba Mon Sep 17 00:00:00 2001
From: Jacky Chou <jacky_chou@aspeedtech.com>
Date: Thu, 7 May 2026 17:44:54 +0800
Subject: [PATCH] arm: mach-aspeed: ast2705: add recovery strap bits

Add AST2705 IO strap definitions for recovery mode and MAC boot
selection. These bits are used to identify when recovery should boot
from MAC instead of normal storage media.

Upstream-Status: Backport [SDK v11.03 migration]
Signed-off-by: Jacky Chou <jacky_chou@aspeedtech.com>
Change-Id: I1b4627308a28ae28251bfaa7d41b469338346c82
Signed-off-by: rupeshs <rupeshs@ami.com>
---
 arch/arm/include/asm/arch-aspeed/scu_ast2700.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/include/asm/arch-aspeed/scu_ast2700.h b/arch/arm/include/asm/arch-aspeed/scu_ast2700.h
index 59e73fd3c2..38ca9a8de7 100644
--- a/arch/arm/include/asm/arch-aspeed/scu_ast2700.h
+++ b/arch/arm/include/asm/arch-aspeed/scu_ast2700.h
@@ -69,8 +69,10 @@
 
 /* SoC1 SCU Register */
 #define SCU_IO_HWSTRAP_UFS			BIT(23)
+#define SCU_IO_HWSTRAP_MAC			BIT(17)		/* AST2705 */
 #define SCU_IO_HWSTRAP_EMMC			BIT(11)
 #define SCU_IO_HWSTRAP_SECBOOT			BIT(5)
+#define SCU_IO_HWSTRAP_RECOVERY			BIT(4)
 #define SCU_IO_HWSTRAP_LTPI0_EN			BIT(3)
 #define SCU_IO_HWSTRAP_LTPI1_EN			BIT(1)
 
-- 
2.43.0

